src/apps/vpn/addons/addon.cpp

```diff
@@ -412,60 +413,62 @@ Addon* Addon::create(QObject* parent, const QString& manifestFileName) {
 }
 
 Addon::Addon(QObject* parent, const QString& manifestFileName,
-             const QString& id, const QString& name, const QString& type)
+             const QJsonObject& manifest)
     : QObject(parent),
       m_manifestFileName(manifestFileName),
-      m_id(id),
-      m_name(name),
-      m_type(type) {
+      m_id(manifest["id"].toString()),
+      m_name(manifest["name"].toString()),
+      m_type(manifest["type"].toString()),
+      m_state(manifest["state"].toObject()) {
```

Comment:
For instance you don't check this object at all... This is fine, but inconsistent